scripts/i.spectral/i.spectral.py

```diff
@@ -92,7 +92,7 @@ def write2textf(what, output):
     outf = open(output, "w")
     i = 0
     for row in enumerate(what):
-        i = i + 1
+        i += 1
         outf.write("%d, %s\n" % (i, row))
```

Comment:
Here it's not related to this issue specifically, but I don't undestand why enumerate is used here, but the count it provides isn't used here. (Enumerate is to go through the iterable, but also bundle the index in a tuple, by default the start argument is at 0). And the loop also keeps track of an `i` variable. Are we sure that the output that is written to the file is what is expected? It writes `i` as a `%d` (a number), but then `row` as a `%s` (a string). Without running it, it should then be the string representation of the `row`, that is in fact here a tuple of the count and the item of `what`. Is it expected? Where is it used?

Reply:
To me it seems like a glitch, when the script was ported from shell to Python.

There are no examples or tests that can be used as a reference for how the textfile output is supposed to look like. With the current code it will look something like this:

```
1, (0, [1, 1, 255, 45, 65])
2, (1, [2, 2, 255, 233, 165])
3, (2, [3, 3, 125, 133, 155])
4, (3, [4, 4, 167, 34, 67])
```
That does not seem natural to me as I would expect rather CSV like text output... 

I would likely go for something like:

```
Path(output).write_text("\n".join([f"{i},{','.join(map(str, row))}"for i, row in enumerate(what, 1)]) + "\n")
```
Unless there is a reason for the current output format...

Reply:
@neteler Do you have some more historical context, or an opinion here?

Reply:
The commit preceding the removal of the shell script version is https://github.com/OSGeo/grass/blob/3d692d1df5b57ddec09ca3ebd324b23e04baa03f/scripts/i.spectral/i.spectral

The author named in the docs is @neteler in 1998

The text file output was added by @YannChemin in 2015 in https://github.com/OSGeo/grass/commit/3107c3a8ccdf7ebf2745161594b233b222eea5ab

Reply:
@ninsbl proposal sounds reasonable to me (and yeah, a output test would be very welcome).

Reply:
I'm for using `i, row` with enumerate and outputting just that. That looks like the intention in the original code. I'm not sure how this compares to the one-liner.

Reply:
Either way it will be in a separate PR, it is not something that blocks this PR

Reply:
Created #4160 to track this